### ISSUE TITLE
Probabilities are NoneType when doing supervised learning

### ISSUE THREAD

**Issue by ngjimenez** [MaartenGr/BERTopic#1808]
I'm following the guide [here](https://maartengr.github.io/BERTopic/getting_started/supervised/supervised.html) on how to use BERTopic for supervised learning on the `20newsgroups` dataset. I want to see all calculated probabilities so I added `calculate_probabilities=True` to the BERTopic instance, see below. However, `probs` comes up as `NoneType`. Is some dependency I am lacking or something wrong with my implementation?

`probs` does return an array of probabilities when doing unsupervised learning, so I wonder if this issue is unique to supervised learning. 

```
# Skip over dimensionality reduction, replace cluster model with classifier,
# and reduce frequent words while we are at it.
empty_dimensionality_model = BaseDimensionalityReduction()
clf = LogisticRegression()
ctfidf_model = ClassTfidfTransformer(reduce_frequent_words=True)

# Create a fully supervised BERTopic instance
topic_model= BERTopic(
        umap_model=empty_dimensionality_model,
        hdbscan_model=clf,
        ctfidf_model=ctfidf_model,
        calculate_probabilities = True
)

topics, probs = topic_model.fit_transform(docs, y=y)
```

**Comment by MaartenGr** [comment]
That is correct. The model used to generate the probabilities is by default derived from HDBSCAN. However, since you are using a classification algorithm instead, these probabilities are extracted using a different method and therefore not supported. 

Having said that, to get the probabilities, you could either access the underlying classifier and access its internal attributes which might contain the probabilities or run `.transform` after you have saved and loaded BERTopic with `pytorch` or `safetensors`. The latter generates the probabilities using the cosine similarities between document and topic embeddings and ignores the classifier.

**Comment by jygenius** [comment]
> I'm following the guide [here](https://maartengr.github.io/BERTopic/getting_started/supervised/supervised.html) on how to use BERTopic for supervised learning on the `20newsgroups` dataset. I want to see all calculated probabilities so I added `calculate_probabilities=True` to the BERTopic instance, see below. However, `probs` comes up as `NoneType`. Is some dependency I am lacking or something wrong with my implementation?
> 
> `probs` does return an array of probabilities when doing unsupervised learning, so I wonder if this issue is unique to supervised learning.
> 
> ```
> # Skip over dimensionality reduction, replace cluster model with classifier,
> # and reduce frequent words while we are at it.
> empty_dimensionality_model = BaseDimensionalityReduction()
> clf = LogisticRegression()
> ctfidf_model = ClassTfidfTransformer(reduce_frequent_words=True)
> 
> # Create a fully supervised BERTopic instance
> topic_model= BERTopic(
>         umap_model=empty_dimensionality_model,
>         hdbscan_model=clf,
>         ctfidf_model=ctfidf_model,
>         calculate_probabilities = True
> )
> 
> topics, probs = topic_model.fit_transform(docs, y=y)
> ```

Hi, I am running into the same issue. Did you solve it?
Thank you!
